lib/flutter_sound.dart

```diff
@@ -1,73 +1,82 @@
 import 'package:flutter_sound/android_encoder.dart';
 import 'package:flutter_sound/ios_quality.dart';
-import 'dart:io' show Platform;
 
 // this enum MUST be synchronized with fluttersound/AudioInterface.java  and ios/Classes/FlutterSoundPlugin.h
-enum t_CODEC
-{
-	DEFAULT,
-	CODEC_AAC,
-	CODEC_OPUS,
-	CODEC_CAF_OPUS, // Apple encapsulates its bits in its own special envelope : .caf instead of a regular ogg/opus (.opus). This is completely stupid, this is Apple.
-	CODEC_MP3,
-	CODEC_VORBIS,
-	CODEC_PCM,
+enum t_CODEC {
+  DEFAULT,
+  CODEC_AAC,
+  CODEC_OPUS,
+  CODEC_CAF_OPUS, // Apple encapsulates its bits in its own special envelope : .caf instead of a regular ogg/opus (.opus). This is completely stupid, this is Apple.
+  CODEC_MP3,
+  CODEC_VORBIS,
+  CODEC_PCM,
 }
 
-enum t_AUDIO_STATE
-{
-        IS_STOPPED,
-        IS_PAUSED,
-        IS_PLAYING,
-        IS_RECORDING,
+enum t_AUDIO_STATE {
+  IS_STOPPED,
+  IS_PAUSED,
+  IS_PLAYING,
+  IS_RECORDING,
 }
 
-const defaultPath = 'sound.aac';
-
-final List<String> paths =
-  [
-  		defaultPath,	// DEFAULT
-  		'sound.aac',	// CODEC_AAC
-  		'sound.opus',	// CODEC_OPUS
-  		'sound.caf',	// CODEC_CAF_OPUS
-  		'sound.mp3',	// CODEC_MP3
-  		'sound.ogg',	// CODEC_VORBIS
-  		'sound.wav',	// CODEC_PCM
-];
-
 class FlutterSound {
   static const MethodChannel _channel = const MethodChannel('flutter_sound');
   static StreamController<RecordStatus> _recorderController;
   static StreamController<double> _dbPeakController;
   static StreamController<PlayStatus> _playerController;
+  static StreamController<PlaybackState> _playbackStateChangedController;
+
   /// Value ranges from 0 to 120
   Stream<double> get onRecorderDbPeakChanged => _dbPeakController.stream;
   Stream<RecordStatus> get onRecorderStateChanged => _recorderController.stream;
   Stream<PlayStatus> get onPlayerStateChanged => _playerController.stream;
+
+  /// Notifies the listeners whenever the playback state of the audio player changes.
+  ///
+  /// This stream stops working when releaseMediaPlayer() is called.
+  Stream<PlaybackState> get onPlaybackStateChanged =>
+      _playbackStateChangedController.stream;
+
   @Deprecated('Prefer to use audio_state variable')
   bool get isPlaying => _isPlaying();
   bool get isRecording => _isRecording();
   t_AUDIO_STATE get audioState => _audio_state;
 
-  bool _isRecording() => _audio_state == t_AUDIO_STATE.IS_RECORDING ;
+  /// The current state of the playback
+  PlaybackState _playbackState;
+
+  // Whether the handler for when the user tries to skip forward was set
+  bool _skipTrackForwardHandlerSet = false;
+  // Whether the handler for when the user tries to skip backward was set
+  bool _skipTrackBackwardHandlerSet = false;
+
+  // The handlers for when a Dart method is invoked from the native code
+  Map<String, Function(MethodCall)> _callHandlers =
+      <String, Function(MethodCall)>{};
+
+  bool _isRecording() => _audio_state == t_AUDIO_STATE.IS_RECORDING;
```

Comment:
_audio_state => _audioState

Reply:
The state of the player is indicated also by ```_playbackState```, which is updated from the native methods and gets updated also when the state is changed from the notification. For this reason I believe that ```_playbackState``` is more efficient than ```_audio_state``` and can replace it.
So, I propose to completely delete the ```_audio_state```variable and add the "recording" state to the ```PlaybackState``` ```enum```.